op-node/rollup/driver/state.go

```diff
@@ -278,11 +280,22 @@ func (s *Driver) eventLoop() {
 				return
 			}
 			if s.network != nil && payload != nil {
-				// Publishing of unsafe data via p2p is optional.
-				// Errors are not severe enough to change/halt sequencing but should be logged and metered.
-				if err := s.network.PublishL2Payload(s.driverCtx, payload); err != nil {
-					s.log.Warn("failed to publish newly created block", "id", payload.ID(), "err", err)
-					s.metrics.RecordPublishingError()
+				isLeader, err := s.sequencerConductor.Leader(s.driverCtx)
+				if err != nil {
+					s.log.Error("failed to determine if this node is the leader sequencer, skipping publishing of unsafe data", "err", err)
+				} else if isLeader {
+					if err := s.sequencerConductor.CommitUnsafePayload(s.driverCtx, payload); err != nil {
+						s.log.Error("failed to commit unsafe payload to conductor log", "id", payload.ID(), "err", err)
+					} else {
+						// Publishing of unsafe data via p2p is optional.
+						// Errors are not severe enough to change/halt sequencing but should be logged and metered.
+						if err := s.network.PublishL2Payload(s.driverCtx, payload); err != nil {
+							s.log.Warn("failed to publish newly created block", "id", payload.ID(), "err", err)
+							s.metrics.RecordPublishingError()
+						}
+					}
+				} else {
+					s.log.Warn("Not the leader sequencer, skipping publishing of unsafe data")
```

Comment:
Just thought about it a little bit more, it seems like our implementation could be reduced to below, commit unsafe payload will inherently check if the current node is leader or not, if it's not a leader anymore, it will return an `ErrNotLeader` error.

And we don't need to check if is leader after this, because after `CommitUnsafePayload` finished, it would be safe to gossip the block to the network, even if sequencer fails here, the backup sequencer will be able to get the block from consensus

```suggestion
// CommitUnsafePayload will return error if current sequencer is not leader now
if err := s.sequencerConductor.CommitUnsafePayload(s.driverCtx, payload); err != nil {
    s.log.Error("failed to commit unsafe payload to conductor log", "id", payload.ID(), "err", err)
    continue
}

// Publishing of unsafe data via p2p is optional.
// Errors are not severe enough to change/halt sequencing but should be logged and metered.
if err := s.network.PublishL2Payload(s.driverCtx, payload); err != nil {
    s.log.Warn("failed to publish newly created block", "id", payload.ID(), "err", err)
    s.metrics.RecordPublishingError()
}
```